src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -318,12 +350,17 @@ public void BuildCFIMap(ObjectNode node)
 
                 if (_targetPlatform.OperatingSystem == TargetOS.Windows)
                 {
-                    SwitchSection(_nativeObjectWriter, "xdata");
-                    EmitAlignment(4);
-
-                    // TODO: Make sure that the same blobs get folded by emitting them to comdat sections
-
                     string blobSymbolName = "_unwind" + (i++).ToStringInvariant() + _currentNodeName;
+
+                    ObjectNodeSection section = ObjectNodeSection.XDataSection;
+                    if (node.ShouldShareNodeAcrossModules(context) && context.Target.OperatingSystem == TargetOS.Windows)
```

Comment:
If there's a cost associated with putting things in a custom section, how about the nodes that implement ShouldShare as "return true"?

I think what would work is:
* Remove ShouldShare from CompilationGroup
* Make ShouldShare (maybe come up with a better name) a property on the nodes (no need to pass the NodeFactory anymore) and make the logic to compute the value local to the node (irrespective of whether we're doing single file or multifile - if there's a world where this can be shared, it should return true).
* In object writer, use ShouldShare, along with some other flag to control whether we should be emitting it into a comdat foldable section